app/components/authors_and_editors.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Components
+  # Component for rendering authors and editors metadata on show pages.
+  #
+  # Displays authors and editors with appropriate links for requesting
+  # authorship credit or reviewing authors (for admins).
+  #
+  # @example For description objects
+  #   render(Components::AuthorsAndEditors.new(
+  #     obj: @name_description,
+  #     versions: @versions,
+  #     user: @user
+  #   ))
+  #
+  # @example For non-description versioned objects
+  #   render(Components::AuthorsAndEditors.new(
+  #     obj: @name,
+  #     versions: @versions,
+  #     user: @user
+  #   ))
+  #
+  class AuthorsAndEditors < Base
+    prop :obj, _Any # Any object with type_tag
+    prop :versions, _Union(Array, ActiveRecord::Associations::CollectionProxy),
+         default: -> { [] }
+    prop :user, _Nilable(User)
+
+    def view_template
+      type = @obj.type_tag
+
+      authors, editors = if description_object?(type)
+                           description_authors_and_editors
+                         else
+                           non_description_authors_and_editors
+                         end
+
+      content = (authors || view_context.safe_empty) +
+                view_context.safe_br +
+                (editors || view_context.safe_empty)
+      p { trusted_html(content) }
+    end
+
+    private
+
+    def description_object?(type)
+      /description/.match?(type.to_s)
+    end
+
+    # Renders authors and editors for description objects
+    def description_authors_and_editors
+      authors_list = @obj.authors
+      editors_list = @obj.editors
+      is_admin = @user && @obj.is_admin?(@user)
+      is_author = @user && authors_list.include?(@user)
+
+      authors = view_context.user_list(:show_name_description_author,
+                                       authors_list)
+      editors = view_context.user_list(:show_name_description_editor,
+                                       editors_list)
+
+      if is_admin
+        authors = authors_with_review_link(authors)
+      elsif !is_author
+        authors = authors_with_request_link(authors)
+      end
+
+      [authors, editors]
+    end
+
+    def authors_with_review_link(authors)
+      return authors if authors.nil? || authors.to_s.empty?
+
+      authors + view_context.safe_nbsp + link_to(
+        "(#{:review_authors_review_authors.t})",
+        description_authors_path(id: @obj.id, type: @obj.type_tag)
+      )
+    end
+
+    def authors_with_request_link(authors)
+      return authors if authors.nil? || authors.to_s.empty?
+
+      authors + view_context.safe_nbsp + link_to(
```

Comment:
If it doesn't really need to be a non-breaking space, Phlex `whitespace` should work.